### PR TITLE
ASG EC2 Auto EIP

### PR DESCRIPTION
**Pull request by igorkotof**

### CODE REVIEW

File: data.tf [hunk 1]
```diff
@@ -9,5 +9,11 @@ data "template_file" "asg_ecs_ec2_user_data" {
     ecs_cluster_name  = local.ecs_cluster_name
     env               = var.env
     ec2_service_group = var.ec2_service_group
+    auto_assign_eip   = var.ec2_asg_auto_assign_eip
```
Comment (DimmKirr) [inline data.tf]: Can we use https://www.terraform.io/language/functions/tostring here and switch the whole var to bool?

File: autoscaling.tf [hunk 2]
```diff
@@ -55,3 +65,23 @@ module "autoscaling" {
...
+        Action = [
+          "ec2:Describe*",
+          "ec2:AssociateAddress"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
```
Comment (DimmKirr) [inline autoscaling.tf]: I'd prefer a new line here please

File: ecs_ec2_user_data.sh.tpl [hunk 3]
```diff
@@ -45,3 +45,110 @@ echo "*/5 * * * * root /etc/ecs/cw_ecsmetrics.sh" > /etc/cron.d/ecsmetrics
 } >> /etc/ecs/ecs.config
 start ecs
 echo "Done"
+
+# ASG Auto Assign Elastic IP
+TIMEOUT=20
+PAUSE=5
+
+apt-get update
```
Comment (DimmKirr) [inline ecs_ec2_user_data.sh.tpl]: As far as I know AWS ECS optimized AMI is is RHEL based, any reason to use ubuntu for that?

File: ecs_ec2_user_data.sh.tpl [hunk 4]
```diff
@@ -45,3 +45,110 @@ echo "*/5 * * * * root /etc/ecs/cw_ecsmetrics.sh" > /etc/cron.d/ecsmetrics
...
+apt install -y curl awscli
+
+aws_get_instance_id() {
+	instance_id=$( (curl http://169.254.169.254/latest/meta-data/instance-id) )
```
Comment (DimmKirr) [inline ecs_ec2_user_data.sh.tpl]: Any reason why there are double braces?

File: autoscaling.tf [hunk 5]
```diff
@@ -1,3 +1,13 @@
+resource "aws_eip" "autoscaling" {
+  count             = var.ec2_asg_auto_assign_eip == "true" ? var.max_size + 1 : 0                             # CHANGE ME (Should be max_size of ASG + 1 )
+  vpc               = false
+  public_ipv4_pool  = "amazon"
+
+  tags = {
```
Comment (DimmKirr) [inline autoscaling.tf:6]: Should we add some sequential id here?

File: variables.tf [hunk 6]
```diff
@@ -366,6 +366,12 @@ variable "autoscale_scheduled_down" {
   default     = []
 }
 
+variable "ec2_asg_auto_assign_eip" {
+  type        = string
```
Comment (DimmKirr) [inline variables.tf]: If we use `tostring()` in the template, we could switch to bool here, I think

**Review by DimmKirr — CHANGES_REQUESTED** [review]
Looks good, thank you. There are some suggestions though.

File: autoscaling.tf (hunk 5 above)
Comment (igorkotof) [reply autoscaling.tf:6]: it's not needed @AutomationD

**Pull request merged**